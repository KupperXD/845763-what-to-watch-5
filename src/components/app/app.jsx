import React from "react";
import Home from "../home/home";
import AddReview from "../add-review/add-review";
import Film from "../film/film";
import MyList from "../my-list/my-list";
import Player from "../player/player";
import SignIn from "../sign-in/sign-in";
import {BrowserRouter, Switch, Route} from "react-router-dom/";
import PropTypes from "prop-types";

const App = (props) => {
  const {films} = props;

  return (
    <BrowserRouter>
      <Switch>
        <Route path="/" exact>
          <Home
            films={films}
          />
        </Route>
        <Route path="/login/" exact>
          <SignIn />
        </Route>
        <Route path="/mylist" exact>
          <MyList
            film={films[0]}
          />
        </Route>
        <Route
          path="/films/:id"
          exact
          /* eslint-disable-next-line no-shadow */
          render={(props) => {
            // eslint-disable-next-line react/prop-types
            const id = props.match.params.id;

            const currentFilm = films.find((film) => film.id === id);

            return (
              <Film
                film={currentFilm}
              />);
          }}
        />
        <Route
          path="/films/:id/review"
          exact
          /* eslint-disable-next-line no-shadow */
          render={(props) => {
            // eslint-disable-next-line react/prop-types
            const id = props.match.params.id;
            const currentFilm = films.find((film) => film.id === id);

            return (
              <AddReview
                film={currentFilm}
              />);
          }}
        />
        <Route
          path="/player/:id"
          exact
          /* eslint-disable-next-line no-shadow */
          render={(props) => {
            // eslint-disable-next-line react/prop-types
            const id = props.match.params.id;
            const currentFilm = films.find((film) => film.id === id);

            return (
              <Player
                film={currentFilm}
              />);
          }}
        />
      </Switch>
    </BrowserRouter>
  );
};

App.propTypes = {
  films: PropTypes.array.isRequired,
};

export default App;
