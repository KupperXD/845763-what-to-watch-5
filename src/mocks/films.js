export default [
  {
    name: `The Gentlemen`,
    genre: `comedy`,
    picture: `https://pravorub.ru/upload/content/2020/02/27/e0478da4ae53762e425bc45fe2e2008a.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    id: 0,
    year: `2019`,
    description: `An American expat tries to sell off his highly profitable marijuana empire in London,
        triggering plots, schemes, bribery and blackmail in an attempt to steal his domain out from under him.`,
    director: `Guy Ritchie`,
    starring: [`Matthew McConaughey`, `Charlie Hunnam`, `Michelle Dockery`],
    rating: {
      score: `9,0`,
      level: `Very good`,
      count: 220,
    }
  },
  {
    name: `Pulp Fiction`,
    genre: `thriller`,
    picture: `https://www.ivi.ru/titr/uploads/2016/05/12/8518671426896744564b2a948d1bad90.jpg/1400x0`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    id: 1,
    year: `1994`,
    description: `The lives of two mob hitmen, a boxer, a gangster and his wife, and a pair of diner bandits
        intertwine in four tales of violence and redemption.`,
    director: `Quentin Tarantino`,
    starring: [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
    rating: {
      score: `9,0`,
      level: `Very good`,
      count: 220,
    }
  },
  {
    name: `The Fast and the Furious`,
    genre: `thriller`,
    picture: `https://cs8.pikabu.ru/post_img/big/2017/12/27/6/1514363305143710026.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    id: 2,
    year: `2001`,
    description: `Los Angeles police officer Brian O'Conner must decide where his loyalty
        really lies when he becomes enamored with the street racing world he has been sent undercover to destroy.`,
    director: `Rob Cohen`,
    starring: [`Vin Diesel`, `Paul Walker`, `Michelle Rodriguez`],
    rating: {
      score: `9,0`,
      level: `Very good`,
      count: 220,
    }
  },
];
