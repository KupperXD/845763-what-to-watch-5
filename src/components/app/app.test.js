import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import films from "../../mocks/films";
import user from "../../mocks/user";

import {App} from './app.jsx';


describe(`Render connected to store component`, () => {
  const mockStore = configureStore([]);
  let store = null;
  let appComponent = null;

  beforeEach(() => {
    store = mockStore({
      DATA: {
        films,
        startFilms: films,
        promoFilm: films[0],
        genre: `all`,
      },
      USER: {
        userData: user,
      },
    });
    store.dispatch = jest.fn();

    appComponent = renderer.create(
        <Provider store={store}>
          <App films={films}/>
        </Provider>
    );
  });

  it(`renders correctly`, () => {
    expect(appComponent.toJSON()).toMatchSnapshot();
  });
});

