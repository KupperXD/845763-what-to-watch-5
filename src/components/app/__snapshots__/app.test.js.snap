// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render connected to store component renders correctly 1`] = `
Array [
  <section
    className="movie-card"
    style={
      Object {
        "backgroundColor": "#73B39A",
      }
    }
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Bronson"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--undefined"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <h1
        className="page-title user-page__title"
      />
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <a
            href="/mylist"
            onClick={[Function]}
          >
            <img
              alt="User avatar"
              height="63"
              src="img/1.png"
              width="63"
            />
          </a>
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="Bronson poster"
            height="327"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Bronson
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Action
            </span>
            <span
              className="movie-card__year"
            >
              2008
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/1"
              onClick={[Function]}
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                }
              }
              type="button"
            >
              <svg
                height="14"
                viewBox="0 0 18 14"
                width="18"
              >
                <use
                  href="#in-list"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item undefined"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item undefined"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Action
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <a
            className="small-movie-card__link"
            href="/films/1"
            onClick={[Function]}
          >
            <video
              autoPlay={false}
              controls={false}
              muted={true}
              poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg"
              style={
                Object {
                  "maxWidth": "100%",
                  "objectFit": "cover",
                }
              }
            >
              <source
                src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
                type="video/webm"
              />
            </video>
            <h3
              className="small-movie-card__title"
            >
              Bronson
            </h3>
          </a>
        </article>
      </div>
      <div
        className="catalog__more"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <button
          className="catalog__button"
          onClick={[Function]}
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
